lib/Dialect/FIRRTL/Transforms/CheckCombCycles.cpp

```diff
@@ -358,7 +358,23 @@ class CombGraphIterator
           return InstanceNodeIterator(instance, node, end);
         })
         .Case<SubfieldOp>([&](SubfieldOp subfield) {
-          return SubfieldNodeIterator(subfield, node, end);
+          if (isa_and_nonnull<MemOp>(subfield.getInput().getDefiningOp()))
+            return static_cast<variant_iterator>(
+                SubfieldNodeIterator(subfield, node, end));
+          // This is required to explicitly ignore self initialization connects
+          // for registers.
+          if (isa_and_nonnull<RegOp, RegResetOp>(
+                  getFieldRefFromValue(subfield).getDefiningOp()))
+            return static_cast<variant_iterator>(NodeIterator(node, true));
+          return static_cast<variant_iterator>(NodeIterator(node, end));
+        })
+        .Case<SubindexOp>([&](SubindexOp sub) {
+          // This is required to explicitly ignore self initialization connects
+          // for registers.
```

Comment:
nit: probably "self loops of registers" would be proper word here.

Reply:
Updated the comment